src/compat.rs

```diff
@@ -277,6 +277,14 @@ impl Compatibility {
     }
 }
 
+pub(crate) mod private {
+    use crate::CompatLevel;
+
+    pub trait OptionCompatLevelMut {
+        fn as_option_compat_level_mut(&mut self) -> &mut Option<CompatLevel>;
+    }
+}
```

Comment:
> This feels like it will be a compiler error at some point in the future.

What exactly and why?

Reply:
The `OptionCompatLevelMut` is part of the public API as a constraint on `Compatible`, right? But it's not actually made public.

There is a private type in public interface warning in rustc, but I suppose it doesn't apply to this (yet?).

It's possible this will never get changed since it might be considered a "feature" to prevent others from implementing a trait defined in your library, but it still seems not ideal to me.

Reply:
That may be a hack but there is no other way to make this work without exposing internal APIs. I did use this trick some years ago and it's still working the same. I don't think Rust will change that without an alternative.

Reply:
It's up to you, but if I had to choose between the two solutions I'd tell downstream to just deal with it rather than introducing a hacky workaround like this. Especially since you'll also have to do this for every new `AsRef` thing you want to add in the future.

Reply:
Yep, that's not optimal, but we'll see if we need to add back the `AsMut` API.